File: app/src/main/java/com/project/scratchstudio/kith_andoid/CustomViews/CustomFont.java
package com.project.scratchstudio.kith_andoid.CustomViews;

import android.content.Context;
import android.graphics.Typeface;

import java.util.Locale;

enum CustomFont {
    JURA("fonts/JuraBook.ttf"),
    INTRO("fonts/Intro-Book.otf"),
    INTRO_REGULAR("fonts/intro_regular.ttf"),
    HATTEN("fonts/HATTEN.TTF"),
    ITCEDSCR("fonts/ITCEDSCR.TTF");

    private final String fileName;

    CustomFont(String fileName) {
        this.fileName = fileName;
    }

    static CustomFont fromString(String fontName) {
        return CustomFont.valueOf(fontName.toUpperCase(new Locale("ru", "RU")));
    }

    public Typeface asTypeface(Context context) {
        return Typeface.createFromAsset(context.getAssets(), fileName);
    }
}

File: app/src/main/java/com/project/scratchstudio/kith_andoid/custom_views/to_trash/CustomFont.kt
package com.project.scratchstudio.kith_andoid.custom_views.to_trash

import android.content.Context
import android.graphics.Typeface

import java.util.Locale

internal enum class CustomFont constructor(private val fileName: String) {
    JURA("fonts/JuraBook.ttf"),
    INTRO("fonts/Intro-Book.otf"),
    INTRO_REGULAR("fonts/intro_regular.ttf"),
    HATTEN("fonts/HATTEN.TTF"),
    ITCEDSCR("fonts/ITCEDSCR.TTF");

    fun asTypeface(context: Context): Typeface {
        return Typeface.createFromAsset(context.assets, fileName)
    }

    companion object {

        fun fromString(fontName: String): CustomFont {
            return valueOf(fontName.toUpperCase(Locale("ru", "RU")))
        }
    }
}
